src/utils/validators/utils.ts

```diff
@@ -0,0 +1,22 @@
+import { notUndefined } from '../assertions';
+import { UndefinedVariableError } from './errors';
+import { Validator, Variable } from './types';
+
+function readEnv(envName: string): Variable {
+  return {
+    name: envName,
+    value: process.env[envName],
+  };
+}
+
+export function validateEnv(envName: string, validator: Validator): string {
+  return validator.validate(readEnv(envName));
+}
```

Comment:
If you still want to have multiple validatior, is it possible to just have an array here ?

```suggestion
export function validateEnv(envName: string, ...validators: Validator[]): string {
    const result = readEnv(envName);
    for(const validator in validators){
        result = validator.validate(result);
    }
  return result;
}
```

Reply:
Yes it is, I had finally simplify to only one validation but we could keep it to offer the possibility to add multiple validators, thanks for your suggestion !